# **PROJECT:** playstore_app_analysis

**Project Type** - EDA (Exploratory Data Analysis)

**Contribution** - Individual

## **PROJECT SUMMARY:**

This project aims to explore Google Play Store data to uncover actionable insights for app developers, focusing on user engagement, satisfaction, and app success. By leveraging Exploratory Data Analysis (EDA), the study examines attributes like app category, ratings, and user reviews to identify patterns and trends. The goal is to help developers optimize their apps, improve user experiences, and make data-driven decisions. Key challenges include handling missing data and outliers, with visualizations playing a critical role in presenting findings. Ultimately, this analysis will provide valuable insights to guide developers in a highly competitive marketplace.

## **GITHUB LINK:**

Link: https://github.com/kush-agra-soni/play_store_eda.git

## **PROBLEM STATEMENT**


The Playstore App Review Analysis project aims to uncover key factors that drive app success on the Google Play Store by conducting an in-depth Exploratory Data Analysis (EDA). The project will analyze various attributes such as app ratings, categories, user reviews, and pricing models to identify patterns, correlations, and trends. The goal is to derive actionable insights that can help developers optimize their apps, improve user satisfaction, and make informed decisions to enhance app performance in a competitive marketplace.

#### **Define Your Business Objective?**

*Optimizing App Success Rate*

## **General Guidelines** : -  

1.   Well-structured, formatted, and commented code is required.
2.   Exception Handling, Production Grade Code & Deployment Ready Code will be a plus. Those students will be awarded some additional credits.
     
     The additional credits will have advantages over other students during Star Student selection.
       
             [ Note: - Deployment Ready Code is defined as, the whole .ipynb notebook should be executable in one go
                       without a single error logged. ]

3.   Each and every logic should have proper comments.
4. You may add as many number of charts you want. Make Sure for each and every chart the following format should be answered.
        

```
# Chart visualization code
```
            

*   Why did you pick the specific chart?
*   What is/are the insight(s) found from the chart?
* Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

5. You have to create at least 20 logical & meaningful charts having important insights.


[ Hints : - Do the Vizualization in  a structured way while following "UBM" Rule.

U - Univariate Analysis,

B - Bivariate Analysis (Numerical - Categorical, Numerical - Numerical, Categorical - Categorical)

M - Multivariate Analysis
 ]





# **BEGINING:**

## ***1. KNOW YOUR DATA***

### Import Libraries

In [71]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import plotly.graph_objs as go
import missingno as msno
from sklearn.preprocessing import StandardScaler
from scipy import stats

### Dataset Loading

In [72]:
# Load the play_store_data.csv file
play_data = pd.read_csv('play_store_data.csv')
user_data = pd.read_csv('user_reviews.csv')

### Dataset First View

In [73]:
play_data.head(3)

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   

  Reviews  Size    Installs  Type Price Content Rating  \
0     159   19M     10,000+  Free     0       Everyone   
1     967   14M    500,000+  Free     0       Everyone   
2   87510  8.7M  5,000,000+  Free     0       Everyone   

                      Genres      Last Updated Current Ver   Android Ver  
0               Art & Design   January 7, 2018       1.0.0  4.0.3 and up  
1  Art & Design;Pretend Play  January 15, 2018       2.0.0  4.0.3 and up  
2               Art & Design    August 1, 2018       1.2.4  4.0.3 and up

In [74]:
user_data.head(3)

App                                  Translated_Review  \
0  10 Best Foods for You  I like eat delicious food. That's I'm cooking ...   
1  10 Best Foods for You    This help eating healthy exercise regular basis   
2  10 Best Foods for You                                                NaN   

  Sentiment  Sentiment_Polarity  Sentiment_Subjectivity  
0  Positive                1.00                0.533333  
1  Positive                0.25                0.288462  
2       NaN                 NaN                     NaN

### Dataset Rows & Columns count

In [75]:
print('PlayStore =' , play_data.shape)
print('User Reviews =', user_data.shape)

PlayStore = (10841, 13)
User Reviews = (64295, 5)


### Dataset Information

In [76]:
print(play_data.info())
print()
print(user_data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB
None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64295 entries, 0 to 64294
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtyp

#### Duplicate Values

In [77]:
# Dataset Duplicate Value Count
print('PlayStore Duplicates =', len(play_data[play_data.duplicated()]))
print('User Reviews Duplicates =', len(user_data[user_data.duplicated()]))

PlayStore Duplicates = 483
User Reviews Duplicates = 33616


#### Missing Values/Null Values

In [78]:
# Missing Values/Null Values Count
print('PlayStore Missing Values =\n',play_data.isnull().sum())
print()
print('User Reviews Missing Values =\n',user_data.isnull().sum())

PlayStore Missing Values =
 App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

User Reviews Missing Values =
 App                           0
Translated_Review         26868
Sentiment                 26863
Sentiment_Polarity        26863
Sentiment_Subjectivity    26863
dtype: int64


### What did you know about your dataset?

**<u>PLAYSTORE :</u>**
  The dataset, comprising **10,841** entries across **13** columns, is central to our project focused on analyzing app details from the Google Play Store. ```Key columns include App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver and Android Ver.``` This data offers insights into various aspects of apps, such as their names, categories, user ratings, review counts, sizes, and install figures. The Rating column has significant missing values **(1,474)**, while Type, Content Rating, Current Ver, and Android Ver also have a few missing values.

---

 **<u>USER REVIEWS :</u>**
 The dataset contains **64,295** entries across **5** columns, focusing on user reviews for various apps available on the Google Play Store.``` The key columns include App, Translated_Review, Sentiment, Sentiment_Polarity, and Sentiment_Subjectivity.``` This dataset is crucial for understanding user feedback, sentiment, and overall user satisfaction with different apps. However, it has significant missing values: Translated_Review (26,868), and Sentiment, Sentiment_Polarity, Sentiment_Subjectivity.

## ***2. UNDERSTANDING YOUR VARIABLES***

PLAYSTORE VARIABLES:

In [79]:
# Dataset Columns
play_data.columns
# Dataset Describe
play_data.describe(include='all')

App Category       Rating Reviews                Size    Installs  \
count    10841    10841  9367.000000   10841               10841       10841   
unique    9660       34          NaN    6002                 462          22   
top     ROBLOX   FAMILY          NaN       0  Varies with device  1,000,000+   
freq         9     1972          NaN     596                1695        1579   
mean       NaN      NaN     4.193338     NaN                 NaN         NaN   
std        NaN      NaN     0.537431     NaN                 NaN         NaN   
min        NaN      NaN     1.000000     NaN                 NaN         NaN   
25%        NaN      NaN     4.000000     NaN                 NaN         NaN   
50%        NaN      NaN     4.300000     NaN                 NaN         NaN   
75%        NaN      NaN     4.500000     NaN                 NaN         NaN   
max        NaN      NaN    19.000000     NaN                 NaN         NaN   

         Type  Price Content Rating Genres    Last Updated  \
count   10840  10841          10840  10841           10841   
unique      3     93              6    120            1378   
top      Free      0       Everyone  Tools  August 3, 2018   
freq    10039  10040           8714    842             326   
mean      NaN    NaN            NaN    NaN             NaN   
std       NaN    NaN            NaN    NaN             NaN   
min       NaN    NaN            NaN    NaN             NaN   
25%       NaN    NaN            NaN    NaN             NaN   
50%       NaN    NaN            NaN    NaN             NaN   
75%       NaN    NaN            NaN    NaN             NaN   
max       NaN    NaN            NaN    NaN             NaN   

               Current Ver Android Ver  
count                10833       10838  
unique                2832          33  
top     Varies with device  4.1 and up  
freq                  1459        2451  
mean                   NaN         NaN  
std                    NaN         NaN  
min                    NaN         NaN  
25%                    NaN         NaN  
50%                    NaN         NaN  
75%                    NaN         NaN  
max                    NaN         NaN

USER DATA VARIABLES:

In [80]:
# Dataset Columns
user_data.columns
# Dataset Describe
user_data.describe(include='all')

App Translated_Review Sentiment  Sentiment_Polarity  \
count                 64295             37427     37432        37432.000000   
unique                 1074             27994         3                 NaN   
top     Angry Birds Classic              Good  Positive                 NaN   
freq                    320               247     23998                 NaN   
mean                    NaN               NaN       NaN            0.182146   
std                     NaN               NaN       NaN            0.351301   
min                     NaN               NaN       NaN           -1.000000   
25%                     NaN               NaN       NaN            0.000000   
50%                     NaN               NaN       NaN            0.150000   
75%                     NaN               NaN       NaN            0.400000   
max                     NaN               NaN       NaN            1.000000   

        Sentiment_Subjectivity  
count             37432.000000  
unique                     NaN  
top                        NaN  
freq                       NaN  
mean                  0.492704  
std                   0.259949  
min                   0.000000  
25%                   0.357143  
50%                   0.514286  
75%                   0.650000  
max                   1.000000

### Variables Description

PlayStore variables description:

* App: App name.
* Category: App category.
* Rating: Average user rating.
* Reviews: Number of user reviews.
* Size: App size.
* Installs: Number of installations.
* Type: Free or Paid.
* Price: App price.
* Content Rating: Suitable age group.
* Genres: App genres.
* Last Updated: Date of last update.
* Current Ver: Current app version.
* Android Ver: Required Android version.

---

User Reviews variables description:

* App: App name.
* Translated_Review: Translated review content.
* Sentiment: Review sentiment.
* Sentiment_Polarity: Polarity score.
* Sentiment_Subjectivity: Subjectivity score.

### Check Unique Values for each variable.

In [81]:
# Check Unique Values for each variable.
for i in play_data.columns.tolist():
  print("No. of unique values in PlayStore data for ",i,"is",play_data[i].nunique(),".")

print()

for i in user_data.columns.tolist():
  print("No. of unique values in User reviews data for ",i,"is",user_data[i].nunique(),".")

No. of unique values in PlayStore data for  App is 9660 .
No. of unique values in PlayStore data for  Category is 34 .
No. of unique values in PlayStore data for  Rating is 40 .
No. of unique values in PlayStore data for  Reviews is 6002 .
No. of unique values in PlayStore data for  Size is 462 .
No. of unique values in PlayStore data for  Installs is 22 .
No. of unique values in PlayStore data for  Type is 3 .
No. of unique values in PlayStore data for  Price is 93 .
No. of unique values in PlayStore data for  Content Rating is 6 .
No. of unique values in PlayStore data for  Genres is 120 .
No. of unique values in PlayStore data for  Last Updated is 1378 .
No. of unique values in PlayStore data for  Current Ver is 2832 .
No. of unique values in PlayStore data for  Android Ver is 33 .

No. of unique values in User reviews data for  App is 1074 .
No. of unique values in User reviews data for  Translated_Review is 27994 .
No. of unique values in User reviews data for  Sentiment is 3 .
No

##***3. DATA WRANGLING***

**<u> DATA WRANGLING (USER REVIEWS DATA SETS) </u>**

### Data Wrangling Code

In [82]:
# Load the dataset
df = pd.read_csv('user_reviews.csv')

# Display the initial shape of the dataset
print(f"Initial shape: {df.shape}")

Initial shape: (64295, 5)


In [83]:
# Handle missing values:
# 1. Remove rows where Translated_Review, Sentiment, Sentiment_Polarity, or Sentiment_Subjectivity are NaN
df.dropna(subset=['Translated_Review', 'Sentiment', 'Sentiment_Polarity', 'Sentiment_Subjectivity'], inplace=True)

# 2. Replace remaining NaN values in numerical columns (Sentiment_Polarity, Sentiment_Subjectivity) with the mean values
df = df.assign(
    Sentiment_Polarity=df['Sentiment_Polarity'].fillna(df['Sentiment_Polarity'].mean()),
    Sentiment_Subjectivity=df['Sentiment_Subjectivity'].fillna(df['Sentiment_Subjectivity'].mean())
)

# Display the shape of the dataset after processing
print(f"Shape after processing: {df.shape}")

Shape after processing: (37427, 5)


In [84]:
# Define function to classify sentiment based on mean Sentiment_Polarity
def classify_sentiment(polarity):
    if polarity > 0:
        return 'Positive'
    elif polarity < 0:
        return 'Negative'
    else:
        return 'Neutral'

In [85]:
# Group by 'App' and calculate mean for 'Sentiment_Polarity' and 'Sentiment_Subjectivity'
aggregated_df = df.groupby('App').agg({
    'Sentiment_Polarity': 'mean',
    'Sentiment_Subjectivity': 'mean'
}).reset_index()

# Round values to 2 decimal places
aggregated_df['Sentiment_Polarity'] = aggregated_df['Sentiment_Polarity'].round(2)
aggregated_df['Sentiment_Subjectivity'] = aggregated_df['Sentiment_Subjectivity'].round(2)

In [86]:
# Classify sentiment based on mean Sentiment_Polarity
aggregated_df['Sentiment'] = aggregated_df['Sentiment_Polarity'].apply(classify_sentiment)

# Rename the columns for clarity
aggregated_df.columns = ['App', 'Polarity', 'Subjectivity', 'Sentiment']

In [87]:
# Save the aggregated results to a new CSV file
aggregated_df.to_csv('final_review_data.csv', index=False)

print("Aggregated data saved to 'final_review_data.csv'.")

Aggregated data saved to 'final_review_data.csv'.


In [88]:
df = pd.read_csv('final_review_data.csv')
df.head()

App  Polarity  Subjectivity Sentiment
0             10 Best Foods for You      0.47          0.50  Positive
1  104 找工作 - 找工作 找打工 找兼職 履歷健檢 履歷診療室      0.39          0.55  Positive
2                              11st      0.19          0.46  Positive
3        1800 Contacts - Lens Store      0.32          0.59  Positive
4   1LINE – One Line with One Touch      0.20          0.56  Positive

**<u>SOME INSIGHTS FROM USER REVIEWS DATA SET</u>**

In [89]:
# Load the dataset
df = pd.read_csv('final_review_data.csv')

# Summary Statistics
summary_stats = df[['Polarity', 'Subjectivity']].describe()
print("Summary Statistics for Polarity and Subjectivity:")
print(summary_stats)

Summary Statistics for Polarity and Subjectivity:
         Polarity  Subjectivity
count  865.000000    865.000000
mean     0.200208      0.489954
std      0.163867      0.086018
min     -0.670000      0.000000
25%      0.080000      0.450000
50%      0.200000      0.500000
75%      0.300000      0.540000
max      1.000000      0.920000


In [90]:
# Sentiment Distribution
sentiment_distribution = df['Sentiment'].value_counts()
print("\nSentiment Distribution:")
print(sentiment_distribution)


Sentiment Distribution:
Sentiment
Positive    778
Negative     76
Neutral      11
Name: count, dtype: int64


In [91]:
# Load the dataset
df = pd.read_csv('final_review_data.csv')

# Calculate the correlation matrix
correlation_matrix = df[['Polarity', 'Subjectivity']].corr()

# Print the correlation matrix
print("Correlation Matrix:")
print(correlation_matrix)

Correlation Matrix:
              Polarity  Subjectivity
Polarity      1.000000      0.307083
Subjectivity  0.307083      1.000000


---

**<u> DATA WRANGLING (PLAY STORE DATA SETS) </u>**

In [92]:
# Load the dataset
df = pd.read_csv('play_store_data.csv')

In [93]:
# Identify numerical and categorical columns
numerical_cols = ['Rating', 'Reviews', 'Size', 'Installs', 'Price']
categorical_cols = ['Category', 'Type', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']

In [94]:
# Handle Size column separately
def convert_size(size_str):
    if size_str == 'Varies with device':
        return np.nan
    size_str = size_str.upper()
    if 'M' in size_str:
        return float(size_str.replace('M', '').strip())  # Convert to float
    elif 'K' in size_str:
        return float(size_str.replace('K', '').strip()) / 1000  # Convert K to M
    else:
        return np.nan  # If not M or K, return NaN

df['Size'] = df['Size'].apply(convert_size)

# Format 'Size' column to append "M" to the values
def format_size(value):
    return f"{value:.1f}M" if pd.notna(value) else "Varies with device"

df['Size'] = df['Size'].apply(format_size)

In [95]:
# Handle Installs column
def convert_installs(install_str):
    # Check if the install_str is numeric or not
    if install_str.replace(',', '').replace('+', '').isnumeric():
        return float(install_str.replace(',', '').replace('+', ''))  # Remove commas and '+' and convert to float
    else:
        return np.nan  # Return NaN for non-numeric values

df['Installs'] = df['Installs'].apply(convert_installs)

# Format 'Installs' column with commas and '+' sign
def format_installs(value):
    return f"{int(value):,}+" if pd.notna(value) else np.nan

df['Installs'] = df['Installs'].apply(format_installs)

In [96]:
# Handle Price column
def convert_price(price_str, type_str):
    if type_str == 'Free':
        return 0.0
    elif type_str == 'Paid':
        price_str = price_str.replace('$', '')  # Remove dollar sign
        return float(price_str)  # Convert to float
    else:
        return np.nan

df['Price'] = df.apply(lambda row: convert_price(row['Price'], row['Type']), axis=1)

# Format 'Price' column with dollar sign
df['Price'] = df['Price'].apply(lambda x: f"${x:.2f}" if pd.notna(x) else np.nan)

# Replace NaN values in 'Price' column based on 'Type'
paid_mean = df.loc[df['Type'] == 'Paid', 'Price'].str.replace(r'\$', '', regex=True).astype(float).mean()  # Mean of 'Paid' prices
df['Price'] = df['Price'].fillna(f"${paid_mean:.2f}")

In [97]:
# Handle the 'Rating' column: replace NaN with the mean of 'Rating' and round to 1 decimal place
df['Rating'] = df['Rating'].fillna(df['Rating'].mean()).round(1)

In [98]:
# Replace NaN values in categorical columns with the mode (most frequent value)
df = df.apply(lambda col: col.fillna(col.mode()[0]) if col.name in categorical_cols else col)

In [99]:
# Save the cleaned dataset
df.to_csv('final_playstore_data.csv', index=False)
print("Cleaned dataset saved to 'final_playstore_data.csv'.")

Cleaned dataset saved to 'final_playstore_data.csv'.


In [119]:
# Convert all columns to string
df = df.astype(str)

# Verify the data types
print(df.dtypes)

App               object
Category          object
Rating            object
Reviews           object
Size              object
Installs          object
Type              object
Price             object
Content Rating    object
Genres            object
Last Updated      object
Current Ver       object
Android Ver       object
dtype: object


In [120]:
df = pd.read_csv('final_playstore_data.csv')
df.head(3)

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   

  Reviews   Size    Installs  Type  Price Content Rating  \
0     159  19.0M     10,000+  Free  $0.00       Everyone   
1     967  14.0M    500,000+  Free  $0.00       Everyone   
2   87510   8.7M  5,000,000+  Free  $0.00       Everyone   

                      Genres      Last Updated Current Ver   Android Ver  
0               Art & Design   January 7, 2018       1.0.0  4.0.3 and up  
1  Art & Design;Pretend Play  January 15, 2018       2.0.0  4.0.3 and up  
2               Art & Design    August 1, 2018       1.2.4  4.0.3 and up

**<u>SOME INSIGHTS FROM PLAY STORE DATA SET</u>**

In [124]:
# Load the dataset
df = pd.read_csv('final_playstore_data.csv')

# Convert 'Rating' to numeric
df['Rating'] = pd.to_numeric(df['Rating'], errors='coerce')

# Convert 'Reviews' to numeric
df['Reviews'] = pd.to_numeric(df['Reviews'], errors='coerce')

# Convert 'Size' to numeric (handling M and K)
def convert_size(size_str):
    if isinstance(size_str, str):
        size_str = size_str.upper()
        if 'M' in size_str:
            return float(size_str.replace('M', '').strip())
        elif 'K' in size_str:
            return float(size_str.replace('K', '').strip()) / 1000  # Convert K to M
    return np.nan

df['Size'] = df['Size'].apply(convert_size)

# Convert 'Installs' to numeric (removing commas and '+')
def convert_installs(install_str):
    if isinstance(install_str, str):
        install_str = install_str.replace(',', '').replace('+', '')
    return pd.to_numeric(install_str, errors='coerce')

df['Installs'] = df['Installs'].apply(convert_installs)

# Convert 'Price' to numeric (removing '$')
def convert_price(price_str):
    if isinstance(price_str, str):
        price_str = price_str.replace('$', '')
    return pd.to_numeric(price_str, errors='coerce')

df['Price'] = df['Price'].apply(convert_price)

# Now calculate the Pearson Correlation
pearson_corr = df[['Rating', 'Reviews', 'Size', 'Installs', 'Price']].corr(method='pearson')

# Display the correlation matrix
print("Pearson Correlation Matrix:")
print(pearson_corr)

Pearson Correlation Matrix:
            Rating   Reviews      Size  Installs     Price
Rating    1.000000  0.067662  0.077076  0.050794 -0.017082
Reviews   0.067662  1.000000  0.238215  0.643122 -0.009679
Size      0.077076  0.238215  1.000000  0.164787 -0.023001
Installs  0.050794  0.643122  0.164787  1.000000 -0.011704
Price    -0.017082 -0.009679 -0.023001 -0.011704  1.000000


In [126]:
# Load the dataset
df = pd.read_csv('final_playstore_data.csv')

# Convert necessary columns to numeric
df['Rating'] = pd.to_numeric(df['Rating'], errors='coerce')
df['Reviews'] = pd.to_numeric(df['Reviews'], errors='coerce')
df['Size'] = pd.to_numeric(df['Size'].str.replace('M', ''), errors='coerce')
df['Installs'] = pd.to_numeric(df['Installs'].str.replace(',', '').replace('+', ''), errors='coerce')
df['Price'] = pd.to_numeric(df['Price'].str.replace('$', ''), errors='coerce')

# Calculate Spearman Correlation
spearman_corr = df[['Rating', 'Reviews', 'Size', 'Installs', 'Price']].corr(method='spearman')

# Display the correlation matrix
print("Spearman Correlation Matrix:")
print(spearman_corr)

Spearman Correlation Matrix:
            Rating   Reviews      Size  Installs     Price
Rating    1.000000  0.204028  0.086888       NaN  0.050659
Reviews   0.204028  1.000000  0.367119       NaN -0.171050
Size      0.086888  0.367119  1.000000       NaN -0.054109
Installs       NaN       NaN       NaN       NaN       NaN
Price     0.050659 -0.171050 -0.054109       NaN  1.000000


### What all manipulations have you done and insights you found?

####**User Reviews Dataset**
**F**or the user_reviews dataset, I executed several data cleaning and processing steps to enhance the quality and usability of the data. Initially, I loaded the dataset and checked its shape to understand its initial state. To address missing data, I performed two major operations:

1. Removing Rows with Critical Missing Values: I removed any rows where critical columns—Translated_Review, Sentiment, Sentiment_Polarity, or Sentiment_Subjectivity—had missing values. These columns are crucial for sentiment analysis, so retaining rows with these NaNs could lead to inaccurate results.

2. Filling Missing Values in Numerical Columns: For the remaining NaNs in Sentiment_Polarity and Sentiment_Subjectivity, I replaced them with the mean values of their respective columns. This approach ensures that the numerical analyses are not skewed by missing data.

***After addressing missing values, I proceeded with aggregation:***

* Aggregation by App: I grouped the data by the App column and calculated the mean values for Sentiment_Polarity and Sentiment_Subjectivity. This step allowed me to analyze sentiment trends on an app level rather than individual reviews.

* Sentiment Classification: I classified the mean sentiment polarity into three categories: 'Positive', 'Negative', or 'Neutral', based on its value. This classification provides a summarized view of how users generally feel about each app.

* Data Rounding and Saving: I rounded the aggregated mean values to two decimal places for clarity and consistency. The final cleaned and processed data was saved to a new CSV file, final_review_data.csv.



##**Play Store Dataset**
**F**or the play_store_data dataset, the focus was on cleaning and standardizing various columns to ensure accurate and consistent data analysis. The primary manipulations included:

***Handling Missing and Inconsistent Values:***

1. Size Column: I converted the Size values from strings representing sizes (e.g., "15M", "2.5K") into numerical values in megabytes. Sizes listed as "Varies with device" were marked as NaN. I then calculated the mean of valid sizes and formatted the size values to append "M" for consistency.

2. Installs Column: I cleaned the Installs column by removing commas and plus signs and converting the values to floats. The cleaned data was then formatted with commas and a plus sign to maintain readability.

3. Price Column: I converted the Price values based on the app type (Free or Paid). Free apps were set to 0.0, while paid apps had their prices converted to floats and formatted with a dollar sign. Missing values in the Price column were replaced with the mean price of paid apps.

***Filling Missing Values:***

* Rating Column: Missing ratings were replaced with the mean rating, rounded to one decimal place, to ensure that the dataset remains comprehensive and usable for analyses.

* Categorical Columns: For categorical columns, missing values were filled with the mode (most frequent value), ensuring that the dataset remains complete and consistent for categorical analysis.



##***INSIGHTS:***

###**Insights from the User Reviews Dataset**

1. Overall Sentiment by App: By aggregating the sentiment data at the app level, we can identify which apps have the most positive or negative user feedback. This insight helps in understanding user satisfaction trends for different apps.

2. Sentiment Distribution: The classification of apps into 'Positive', 'Negative', or 'Neutral' sentiment categories provides a clear overview of user perceptions. Apps with predominantly positive sentiment may indicate high user satisfaction, whereas those with negative sentiment may highlight potential areas for improvement.

3. Sentiment Trends: Mean sentiment polarity and subjectivity offer a quantitative measure of overall user sentiment. Analyzing these metrics over time or across different apps can reveal patterns or shifts in user sentiment, helping to track the impact of updates or changes.



###**Insights from the Play Store Dataset**

1. App Size and User Preferences: Standardizing the Size column allows for analysis of app sizes relative to user preferences. For example, understanding the average size of popular apps can provide insights into user tolerance for app size and help developers optimize their app's footprint.

2. Install Trends: By converting and formatting the Installs column, we can analyze installation trends and correlate them with other factors like app category, rating, or price. High install counts for certain categories or types of apps can indicate popular trends and user preferences.

3. Pricing Strategy and Impact: With the Price column cleaned and standardized, we can evaluate how pricing affects app popularity and user ratings. For instance, comparing the pricing of paid apps with their ratings and install counts can reveal whether higher-priced apps correlate with better or worse ratings.

4. Rating Distribution: The standardized Rating column allows for detailed analysis of app ratings. Examining the distribution of ratings across different categories or app types can help identify which categories tend to have higher or lower average ratings.

5. Categorical Insights: Filling missing values in categorical columns with modes helps in understanding the most common app types, categories, and other attributes. This can reveal popular app genres and trends in app features or updates.

6. Data Completeness and Consistency: The thorough cleaning process ensures that the dataset is complete and consistent. This enhances the reliability of any further analysis, such as market segmentation, competitive analysis, or predictive modeling.

**:** *Overall, these manipulations have significantly improved the quality of the datasets, enabling more accurate and insightful analyses.*

## ***4. DATA VISUALIZATION, STORYTELLING & EXPERIMENTING WITH CHARTS: UNDERSTAND THE RELATIONSHIPS BETWEEN VARIABLES***

#### Chart - 1

In [101]:
# Chart - 1 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 2

In [102]:
# Chart - 2 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 3

In [103]:
# Chart - 3 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 4

In [104]:
# Chart - 4 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 5

In [105]:
# Chart - 5 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 6

In [106]:
# Chart - 6 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 7

In [107]:
# Chart - 7 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 8

In [108]:
# Chart - 8 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 9

In [109]:
# Chart - 9 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 10

In [110]:
# Chart - 10 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 11

In [111]:
# Chart - 11 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 12

In [112]:
# Chart - 12 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 13

In [113]:
# Chart - 13 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 14 - Correlation Heatmap

In [114]:
# Correlation Heatmap visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

#### Chart - 15 - Pair Plot

In [115]:
# Pair Plot visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

## **5. Solution to Business Objective**

#### What do you suggest the client to achieve Business Objective ?
Explain Briefly.

Answer Here.

# **Conclusion**

Write the conclusion here.

### ***Hurrah! You have successfully completed your EDA Capstone Project !!!***